ci/test_suites/authenticators_k8s/dev/dev_conjur.template.yaml

```diff
@@ -86,6 +86,12 @@ spec:
         name: conjur
         command: ["conjurctl", "server"]
         env:
+        - name: KUBERNETES_SERVICE_HOST
```

Comment:
For context for future readers, remember that the following `KUBERNETES_*` environment variables are automatically added to a Pod (and their containers) by the Kubernetes cluster (they do not appear explicitly in the Kubernetes manifest / Pod description, only in the container environment themselves). In this case, our value defined here overrides `KUBERNETES_SERVICE_HOST` (IP addresses redacted).

```
 KUBERNETES_SERVICE_PORT_HTTPS=443
KUBERNETES_SERVICE_PORT=443
KUBERNETES_PORT_443_TCP=tcp://x.x.x.x:443
KUBERNETES_PORT_443_TCP_PROTO=tcp
KUBERNETES_PORT_443_TCP_ADDR=x.x.x.x
KUBERNETES_SERVICE_HOST=x.x.x.x
KUBERNETES_PORT=tcp://x.x.x.x:443
KUBERNETES_PORT_443_TCP_PORT=443
```

> NOTE: the default value for KUBERNETES_SERVICE_HOST is actually an IP address, as opposed to a
> DNS record, which caused some confusion around tests with NO_PROXY. An IP address and some arbitrary
> DNS record are not treated as equal entities (or resolved) when used in `NO_PROXY` and `https_proxy`.

Example with curl, which also respects the `NO_PROXY` and `https_proxy` environment variables:

```
root@conjur-authn-k8s-668b6c89c-8t99s:/opt/conjur-server# echo $KUBERNETES_SERVICE_HOST
x.x.x.x
root@conjur-authn-k8s-668b6c89c-8t99s:/opt/conjur-server# echo $https_proxy
http://nonexistent_proxy.local/
root@conjur-authn-k8s-668b6c89c-8t99s:/opt/conjur-server# echo $NO_PROXY
kubernetes.default.svc
root@conjur-authn-k8s-668b6c89c-8t99s:/opt/conjur-server# curl -kv https://$KUBERNETES_SERVICE_HOST
* Uses proxy env variable NO_PROXY == 'kubernetes.default.svc'
* Uses proxy env variable https_proxy == 'http://nonexistent_proxy.local'
* Could not resolve proxy: nonexistent_proxy.local
```